contracts/RoundManager.sol

```diff
@@ -0,0 +1,39 @@
+pragma solidity ^0.4.24;
+
+import "zeppelin-solidity/contracts/math/SafeMath.sol";
+
+contract RoundManager {
+  using SafeMath for uint;
+
+  // Round number of the last round
+  uint public lastRound;
+  // Block number of the start of the last round
+  // N.B: not necessarily the block at which initializeRound() was called
+  uint public startOfLastRound;
+
+  uint public roundLength;
+  uint public rateLockDealine;
+
+  constructor() {
+    roundLength = 20;
+    rateLockDealine = 5;
+  }
+
+  function timeSinceBeginningOfLastRound() internal view returns (uint) {
+    return block.number.sub(startOfLastRound);
+  }
+
+  modifier onlyBeforeActiveRoundIsLocked() {
+    require(timeSinceBeginningOfLastRound() < roundLength.sub(rateLockDealine));
+    _;
+  }
+
+  function initializeRound() external {
```

Comment:
Unless this is called every block or *exactly* every `roundLength`, you will have loss of precision with rounds.

Ex: Stacktrace
```
roundLength = 5
lastRound = 0
```

initializeRound called on block 6 
```
blockNumber = 5
currentRound = 1
lastRound = 1
startOfLastRound = 5
```

initializeRound called on block 23
```
blockNumber = 23
currentRound = 4
lastRound = 4
startOfLastRound = 20
```

The round did not start on block 20, it started on 23. This effectively nullifies `roundLength` and leads to rounds of indeterminate length.

Reply:
You are correct. I took inspiration from Livepeer's implementation, and wondered about the same thing. I will ask them directly about this particular choice